src/util/splitSentence.ts

```diff
@@ -1,16 +1,160 @@
 import { once } from './once'
+import { isAbbrev } from './isAbbreviation'
 
 /**
  * Splits given value by special characters.
  */
-export const splitSentence = (value: string) => {
+export const splitSentence = (value: string) : string[] => {
+  // pattern1, single symbol: . ; ! ?
+  // pattern2, multiple symbols: ?! !!! ...
   const mainSplitRegex = /[.;!?]+/g
 
-  /** Checks if the value has no other main split characters except one period at the end. */
+  const spliters = value.match(mainSplitRegex)
+
+  // When it cannot be split by the main spliter, spliter by ','
+  if (!spliters) return value.split(',').filter(s => s !== '').map(s => s.trim())
+
+  /**
+   * Checks if the value has no other main split characters  except one period at the end.
+   * If so, allow split on comma only if there are no main split characters in the value or has only one period at the end.
+   */
   const hasOnlyPeriodAtEnd = once(() => /^[^.;!?]*\.$[^.;!?]*/.test(value.trim()))
 
-  const hasMainSplitCharacters = mainSplitRegex.test(value)
-  // Allow split on comma only if there are no main split characters in the value or has only one period at the end.
-  const seperator = hasMainSplitCharacters && !hasOnlyPeriodAtEnd() ? mainSplitRegex : ','
-  return value.split(seperator).filter(s => s !== '').map(s => `${s.trim()}${hasMainSplitCharacters && !hasOnlyPeriodAtEnd() ? '.' : ''}`)
+  if (hasOnlyPeriodAtEnd()) return value.split(',').filter(s => s !== '').map(s => `${s.trim()}`)
+
+  /**
+   * When the setences can be split, it has multiple situations.
+   */
+  const sentences = value.split(mainSplitRegex)
+
+  /**
+   * The reduce function return a string, which is a combination of all the sentences, we then can split it during the next step.
+   * Google searching keyword '^^' or '^#' return 0 result for both of them. Hence, it is safe in using the below seperator to link each sentence into one string.
+   */
+  const seperator = value.indexOf('^^') > 0 ? '^#' : '^^'
```

Comment:
This is a good idea, but let's use a dummy value like `__SEP__` instead, which is clearer and even more conflict-proof. Define it as a const at the top of the file, like this:

https://github.com/cybersemics/em/blob/66855eafb13b103520aaee9d0a7a4a60f00346fd/src/util/hashContext.ts#L10